packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -104,12 +174,97 @@ export class FluidAppInsightsLogger implements ITelemetryBaseLogger {
 		return shouldSendEvent;
 	}
 
+	/**
+	 * Checks if a given telemetry event conforms to any of the provided {@link TelemetryFilter} rules.
+	 *
+	 * 1. The event must match the requirements of the most specific relevant filter to it. This takes precedence over a more generic filter.
+	 * The less categories and longer the namespace within a filter, the more specific it is. Definining no categories is equivalant to defining all categories.
+	 *
+	 * 2. If a {@link TelemetryFilter} specifies both `categories` and a `namespace`, the event must match both.
+	 *
+	 * 3. If only `categories` or a `namespace` is provided, the event should match either one of them.
+	 *
+	 * 4. If a `namespace` pattern exception is specified in the {@link TelemetryFilter}, the event should not match the exception pattern.
+	 *
+	 * @param event - The telemetry event to check against the filters.
+	 *
+	 * @returns boolean `true` if the event matches any filter, otherwise `false`.
+	 */
 	private doesEventMatchFilter(event: ITelemetryBaseEvent): boolean {
 		for (const filter of this.config.filtering.filters ?? []) {
-			if (filter.category === event.category) {
+			if ("namespacePattern" in filter) {
+				if (event.eventName.startsWith(filter.namespacePattern)) {
+					// Found matching namespace pattern, since filters are ordered in most specific first,
+					// this is the most specific, relevant matching filter for the event.
+
+					// By default, if no categories are defined then any category is a valid match.
+					let doesFilterCategoriesMatch = true;
+					if ("categories" in filter && filter.categories.length > 0) {
+						doesFilterCategoriesMatch = false;
+						const matchingCategory = filter.categories.find(
+							(category) => category === event.category,
+						);
+						doesFilterCategoriesMatch = matchingCategory ? true : false;
+					}
+
+					if (doesFilterCategoriesMatch) {
+						// The most specific, relevant filter matches so no need to attempt to evaluate against other filters
+						// as long as the events namespace does not match any defined namespace exception.
+						if (filter.namespacePatternExceptions !== undefined) {
+							for (const patternException of filter.namespacePatternExceptions) {
+								if (event.eventName.startsWith(patternException)) {
+									return false;
+								}
+							}
+						}
```

Comment:
I'd move this to be the first check once we match on namespace. But note that it cannot `return false`, that prevents any further filters from being evaluated; I think inside the loop we just want to `continue;` so the next filter gets a change to see if it matches the event. That applies to the lines below too.

Reply:
Reviewed with Sean, I misread how this works. It's actually correct.